src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,432 @@
+    const [selectedHour, setSelectedHour] = useState<number | null>();
+
+    const [step, setStep] = useState(0);
+    const [availabilitiesMap, setAvailabilities] = useState<Map<string, number[]>>();
+    const [availableHours, setAvailableHours] = useState<number[]>();
+
+    // Calculates the number of days in the given month
+    const getNoOfDays = useCallback(
+        (month: number) => {
+            let daysInMonth = new Date(year, month + 1, 0).getDate();
+            let dayOfWeek = new Date(year, month).getDay();
+
+            let blankdaysArray: number[] = [];
+            for (var i = 1; i <= dayOfWeek; i++) {
+                blankdaysArray.push(i);
+            }
+
+            let daysArray: number[] = [];
+            for (var i = 1; i <= daysInMonth; i++) {
+                daysArray.push(i);
+            }
+
+            setBlankDays(blankdaysArray);
+            setNumDays(daysArray);
+        },
+        [year],
+    );
+
+    useEffect(() => {
+        getNoOfDays(month);
+    }, [month, getNoOfDays]);
+
+    useEffect(() => {
+        if (availabilities) {
+            const datesList = availabilities.map((d: Availability) => d.attributes.date);
+            const dateMap: Map<string, number[]> = new Map();
+
+            for (let d of datesList) {
+                const date = new Date(d).toLocaleDateString();
+                const hour = parseInt(d.substr(11, 2));
+                if (!dateMap.has(date)) {
+                    dateMap.set(date, []);
+                }
+                const hoursArray = dateMap.get(date);
+                if (hoursArray) hoursArray.push(hour);
+            }
+
+            setAvailabilities(dateMap);
+            setAvailableHours(dateMap.get(new Date().toLocaleDateString()));
+        }
+    }, [availabilities]);
+
+    // Checks if the day is the current day
+    const isToday = (date: number) => {
+        const today = new Date();
+        const d = new Date(year, month, date);
+
+        return today.toDateString() === d.toDateString() ? true : false;
+    };
+
+    // Sets an actual date value from the calendar number passed in
+    const selectDateValue = (day: number) => {
+        let selectedDate = new Date(year, month, day);
+
+        setDatepickerValue(() => selectedDate.toDateString());
+        setDay(day);
+        setSelectedHour(null);
+
+        if (availabilitiesMap) setAvailableHours(availabilitiesMap.get(selectedDate.toLocaleDateString()));
+
+        setStep(1);
+    };
+
+    const selectHour = (hour: number) => {
+        setSelectedHour(hour);
+    };
+
+    const submitDate = () => {
+        if (selectedHour) {
+            const date = new Date(year, month, day, selectedHour);
+            //send this date object through params
+        }
+    };
+
+    const hasNoAvailability = (day: number): boolean => {
+        const dateKey = new Date(year, month, day).toLocaleDateString();
+
+        return !availabilitiesMap?.get(dateKey);
+    };
```

Comment:
ditto

also, whenever messing with dates, never make them a string unless its to show to the user